docs/iot-devkit/devkit-stream-analytics-cosmos-db.md

```diff
@@ -0,0 +1,241 @@
+  float temperature = 0;
+  sensor->getTemperature(&temperature);
+  return temperature;
+}
+
+void setup() {
+  // put your setup code here, to run once:
+  initSensor();
+  if (WiFi.begin() == WL_CONNECTED)
+  {
+    hasWifi = true;
+    Screen.print(1, "Running...");
+
+    if (!DevKitMQTTClient_Init())
+    {
+      hasIoTHub = false;
+      return;
+    }
+    hasIoTHub = true;
+  }
+  else
+  {
+    hasWifi = false;
+    Screen.print(1, "No Wi-Fi");
+  }
+}
+
+void loop() {
+  // put your main code here, to run repeatedly:
+  if (hasIoTHub && hasWifi)
+  {
+    char buff[128];
+    float temperature = readTemperature();
+
+    // replace the following line with your data sent to Azure IoTHub
+    snprintf(buff, 128, "{\"temperature\":%.2f}", temperature);
+    
+    if (DevKitMQTTClient_SendEvent(buff))
+    {
+      Screen.print(1, "Sending...");
+    }
+    else
+    {
+      Screen.print(1, "Failure...");
+    }
+    delay(2000);
+  }
+}
+```
+
+## Provision Azure Service
+
+Use `F1` or `Ctrl+Shift+P` (macOS: `Cmd+Shift+P`) to open the command palette, type **IoT Workbench**, and then select **IoT Workbench: Cloud**, **Azure Provision**.
+
+![Azure Provision](media/iot-workbench-cloud-provision2.png)
+
+Select subscription and resource group you want to create Azure services in. A guide line show what service will be created.
+
+![Azure Provision Step](media/iot-workbench-stream-analytics-and-cosmos-db-provision-step.png)
+
+Follow the guide to create Azure services.
+
+> Notice: Currently, you need specific subscription and resource group for IoT Hub individually.
+
+## Deploy Azure Stream Analytics Job Query
+
+You can change Azure Stream Analytics Job Query by editing `StreamAnalytics/query.asaql`. ASAQL (Azure Stream Analytics Query Language) is SQL-like, you can learn more about it from <https://go.microsoft.com/fwLink/?LinkID=619153>.
+
+![Azure Stream Analytics Query](media/iot-workbench-stream-analytics-and-cosmos-db-query.png)
+
+Output and input have already been generated automatically by default (in the example above, the output is `cosmosdb-a94a5672-867c-6b4e-db41-872d6e01e4bf`, and input is `iothub-ff8feba1-b114-48de-d8c4-d25e7efa4864`). And you have no need to change them. `*` here is the same as SQL, means data in all feilds.
+
+You can select specific feild data from input and export it into output. For example, your device sends data in such JSON format to IoT Hub:
+
+```json
+{
+    "temperature": 42
+}
+```
+
+You can write query as below:
+
+```sql
+SELECT
+    temperature
+FROM
+    "iothub-ff8feba1-b114-48de-d8c4-d25e7efa4864"
+INTO
+    "cosmosdb-a94a5672-867c-6b4e-db41-872d6e01e4bf"
+```
+
+Azure Steam Analytics Query Language (ASAQL) has windowing functions support. A window contains event data along a timeline and enables you to perform various operations against the events within that window. For example, you may want to calculate the average of the values of payload fields in a given window.

```

Comment:
Azure Steam Analytics Query Language (ASAQL)  has native support for windowing functions.    

I suggest you just reference  https://docs.microsoft.com/en-us/azure/stream-analytics/stream-analytics-window-functions